ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Embed/Nary.hs

```diff
@@ -135,7 +137,8 @@ instance Inject GenTx where
   inject _ = injectNS' (Proxy @GenTx)
 
 instance Inject WrapGenTxId where
-  inject _ = injectNS' (Proxy @WrapGenTxId)
+  inject _ ix w = WrapGenTxId $ HardForkGenTxId $ OneEraGenTxId $
+    hcollapse $ hcmap proxySingle (K . toRawTxIdHash . unwrapGenTxId) $ injectNS ix w
```

Comment:
I was briefly confused why we `injectNS` and then immediately `hcollapse` again, but we only need to do this to get an `ConvertRawTxId` dictionary :smile: 

What about unifying the implementation here and in `Inject WrapHeaderHash`?

Reply:
lol yeah I'll change this definition to be the same as WrapHeaderHash, this is needlessly complicating it